src/pybind/mgr/pg_autoscaler/module.py

```diff
@@ -385,6 +428,31 @@ def _maybe_adjust(self):
                     'var': 'pg_num',
                     'val': str(p['pg_num_final'])
                 })
+                
+                # Create new event for each pool
+                # and update existing events
+                # Call Progress Module to create progress event
+                if pool_id not in self._event:
+                    osdmap = self.get_osdmap()
+                    pools = osdmap.get_pools()
+                    pool_data = pools[int(pool_id)]
+                    pg_num = pool_data['pg_num']
+                    pg_num_target = pool_data['pg_num_target']
+                    ev_id = str(uuid.uuid4())
+                    pg_adj_obj = PgAdjustmentProgress(pg_num, pg_num_target, ev_id)
+                    self._event[pool_id] = pg_adj_obj
+                    if pg_num < pg_num_target:
```

Comment:
could pull this into a make_remote_event() helper since it's almost the same as in _update_progress_events(). actually maybe you could just call _update_progress_events() here instead

Reply:
> could pull this into a make_remote_event() helper since it's almost the same as in _update_progress_events(). actually maybe you could just call _update_progress_events() here instead
@jdurgin it similar however, there is a slight difference in that _update_progress_events() iterates through all events in this module and updates the pg_num, however with this block of code, we are checking if there is an event in the pool that needs adjusting and have `pg_autoscale_mode == on`, and we are creating PgAdjustmentProgress() object and deciding if the pg is increasing or decreasing, therefore I don't think we can call _update_progress_event without modifying it to also have these functionalities but if it's okay with you, I think keeping it separate would make more sense.

Reply:
keeping it separate is fine, there's not much repeated now